face/VatomView.js

```diff
@@ -0,0 +1,119 @@
+    // create loader
+    if (this.config.loader) {
+      this.loader = this.config.loader;
+    } else {
+      this.loader = document.createElement('div');
+      this.loader.innerHTML = '<svg width="100%" height="100%" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100" preserveAspectRatio="xMidYMid" class="lds-double-ring"><circle cx="50" cy="50" ng-attr-r="{{config.radius}}" ng-attr-stroke-width="{{config.width}}" ng-attr-stroke="{{config.c1}}" ng-attr-stroke-dasharray="{{config.dasharray}}" fill="none" stroke-linecap="round" r="40" stroke-width="4" stroke="#456caa" stroke-dasharray="62.83185307179586 62.83185307179586" transform="rotate(305.714 50 50)"><animateTransform attributeName="transform" type="rotate" calcMode="linear" values="0 50 50;360 50 50" keyTimes="0;1" dur="1s" begin="0s" repeatCount="indefinite"></animateTransform></circle><circle cx="50" cy="50" ng-attr-r="{{config.radius2}}" ng-attr-stroke-width="{{config.width}}" ng-attr-stroke="{{config.c2}}" ng-attr-stroke-dasharray="{{config.dasharray2}}" ng-attr-stroke-dashoffset="{{config.dashoffset2}}" fill="none" stroke-linecap="round" r="35" stroke-width="4" stroke="#88a2ce" stroke-dasharray="54.97787143782138 54.97787143782138" stroke-dashoffset="54.97787143782138" transform="rotate(-305.714 50 50)"><animateTransform attributeName="transform" type="rotate" calcMode="linear" values="0 50 50;-360 50 50" keyTimes="0;1" dur="1s" begin="0s" repeatCount="indefinite"></animateTransform></circle></svg>';
+    }
+
+    this.update();
+  }
+
+  update() {
+    // Check if ready to be displayed
+    if (!this.vatomObj) {
+      console.warn('No vAtom supplied');
+      // eslint-disable-next-line
+      return;
+      // eslint-disable-next-line
+    } else {
+      this.free();
+      this.load();
+    }
+  }
+
+  load() {
+    // start the face selection procedure
+    const st = this.fsp(this.vatomObj.faces);
+    // check if face is registered
+    const du = st.properties.display_url;
+    let FaceClass = registeredFace[du];
+    // if there is no face registered in the array but we have a http link, show the web face
+    if (FaceClass === undefined && du.indexOf('http') !== -1) {
+      FaceClass = ImageFace;
+    } else if (FaceClass === undefined && du.indexOf('http') === -1) {
+      // display activated Image in the case that the face is undefined!
+      FaceClass = ImageFace;
+    }
+    // create a new instance of the chosen face class and pass through the information
+    const rFace = new FaceClass(this, this.vatomObj, st);
+    this._currentFace = rFace;
+    // make rface opaque
+    rFace.element.style.opacity = 0;
+    // add face to element
+    this.element.appendChild(rFace.element);
+
+    // add the loader
+    this.element.appendChild(this.loader);
+    // call rface.onload , wait for promise
+    Promise.resolve(() => null).then(() => rFace.onLoad()).then(() => {
+      this.element.removeChild(this.loader);
+      rFace.element.style.opacity = 1;
+    });
```

Comment:
This promise needs a `.catch(`, or else the vatomview will just look like it's loading indefinitely if something goes wrong... Maybe just log the error and show the image face again?